app/services/description_headers.rb

```diff
@@ -7,18 +7,25 @@ def self.create(headers:)
   end
 
   def initialize(headers:)
-    @headers = headers
+    # Shift source_id to the front of the headers
+    @headers = ['source_id'] + (headers - ['source_id']).to_a
```

Comment:
Shift the sourceId to the front, and do the conversion from set to array, up front. (Because we don't appear to need the original set.)